compiler/luci/import/src/Importer.cpp

```diff
@@ -315,6 +315,35 @@ std::unique_ptr<Module> Importer::importModule(const circle::Model *model) const
 
   post_import_graph(module.get(), reader);
 
+  // Initialize 'source_table'
```

Comment:
I think the validation is missing. These codes are inserted after origins are already created and there could be differences between the node-ids that are created when they are imported and the one that are from circle model's source table that exists in binary itself.

There are two situations. One is when circle model doesn't have the source table(`circle_metadata->source_table().size() == 0`). And, the second is the one who has it(`circle_metadata->source_table().size() > 0`). Former has no problem because node-ids are generated at the time of the importing model. But, IIUC, latter has a problem. There is no guarantee that node-id that is generated above and the one from `source_table` are same because previous model might have been optimized as what you said.

So, if circle model has `source_table`, these should be guaranteed.
- the number of subgraphs is 1.
- set node id to the one from `source_table` not from its importing order.